ostd/src/arch/x86/poweroff.rs

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Provides the implementation of the poweroff functionality.
+
+use super::qemu::{exit_qemu, QemuExitCode};
+use crate::arch::cpu::cpuid::query_is_running_in_qemu;
+
+/// Powers off the system.
+pub fn poweroff() -> ! {
```

Comment:
This function is not significant enough to be a top-level module of `arch`. I suppose that a more appropriate location for the `poweroff` function is the `cpu` module.